graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/TypeBuilder.kt

```diff
@@ -47,12 +47,12 @@ internal open class TypeBuilder constructor(protected val generator: SchemaGener
 
         // Do not call the hook on GraphQLTypeReference as we have not generated the type yet
         val unwrappedType = GraphQLTypeUtil.unwrapType(graphQLType).lastElement()
+        val typeWithNullability = graphQLType.wrapInNonNull(type)
```

Comment:
This would throw an exception if I had a type returned from the hooks that correctly had it wrapped in a `GraphQLNonNull` right? I think we can refactor the `wrapInNonNull` method to have the logic to check if it is already non-null. If it is just return it instead of throwing an exception

@dariuszkuc What do you think?

Reply:
Not sure what the original purpose of throwing that error was, but it would have thrown an error if it a GraphQLNonNull type wrapped anything besides a GraphQLTypeReference. I think the only ways for a GraphQLNonNull type to get into that method are to create a scalar type that is wrapped in GraphQLNonNull or implement onRewireGraphQLType to wrap a type in GraphQLNonNull. Idk if there is a valid use case for either one of those.

Reply:
I don't think there is. Let's just change the logic and remove the exception and all its test cases

Reply:
ok done.